src/main/java/seedu/address/model/person/FriendName.java

```diff
@@ -18,27 +19,24 @@
      */
     public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
 
-    public final String fullName;
-
     /**
-     * Constructs a {@code Name}.
+     * Constructs a {@code FriendName}.
      *
      * @param name A valid name.
      */
-    public Name(String name) {
-        requireNonNull(name);
-        checkArgument(isValidName(name), MESSAGE_CONSTRAINTS);
-        fullName = name;
+    public FriendName(String name) {
+        super(name);
+        checkArgument(isValidFriendName(name), MESSAGE_CONSTRAINTS);
     }
 
     /**
      * Returns true if a given string is a valid name.
      */
-    public static boolean isValidName(String test) {
+    public static boolean isValidFriendName(String test) {
```

Comment:
A possible alternative is to put the `isValidName` static method in the parent `Name` class, and pass the child object `this` into it. Then the parent static method can check according to class by referencing some sort of static REGEX string for each child class.

Reply:
Even with the alternative, we would have to implement a function in the child class, thus to avoid complexity I think we should stick to individually implementing the isValid[Class]Name. We could put a note in the DG regarding this to ensure that anybody modifying Amigos in the future can take note of this.